packages/core/test/actor.test.ts

```diff
@@ -1177,4 +1177,49 @@ describe('actors', () => {
 
     expect(service.getSnapshot().value).toBe('done');
   });
+
+  it('should not restart a completed observable', (done) => {
+    let subscriptionCount = 0;
+    const machine = createMachine({
+      invoke: {
+        id: 'observable',
+        src: fromObservable(() => {
+          const obs = of(42);
+          const sub = obs.subscribe.bind(obs);
+
+          obs.subscribe = (...args) => {
+            subscriptionCount++;
+            return sub(...(args as any));
+          };
+
+          return obs;
+        })
+      }
+    });
+
+    const actor = interpret(machine).start();
+
+    const obs = actor
+      .getSnapshot()
+      .children['observable'].getPersistedState?.();
+
+    expect(obs).toEqual(
+      expect.objectContaining({
+        data: 42,
+        status: 'done'
+      })
+    );
```

Comment:
This kinda ain't part of this test, I don't mind it that much but I find shorter tests to be easier to read
```suggestion
```